astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -2082,6 +2097,70 @@ def test_unstructured_to_structured(self):
         # For the other tests of ``structured_to_unstructured``, see
         # ``test_structured.TestStructuredQuantityFunctions.test_unstructured_to_structured``
 
+    def test_merge_arrays(self):
+        """Test `numpy.lib.recfunctions.merge_arrays`."""
+        # merge 1 normal array
+        arr = rfn.merge_arrays(self.q_pv["p"])
+        assert np.array_equal(arr.value["f0"], [1, 2, 3])
+        assert arr.unit == (u.km,)
+
+        # merge 1 structured array
+        arr = rfn.merge_arrays(self.q_pv)
+        assert np.array_equal(arr.value["p"], [1, 2, 3])
+        assert np.array_equal(arr.value["v"], [0.25, 0.5, 0.75])
+        assert arr.unit == (u.km, u.km/u.s)
+
+        # merge 1-elt tuple
+        arr = rfn.merge_arrays((self.q_pv,))
+        assert np.array_equal(arr.value["p"], [1, 2, 3])
+        assert np.array_equal(arr.value["v"], [0.25, 0.5, 0.75])
+        assert arr.unit == (u.km, u.km/u.s)
+
+        # merge 2-elt tuples
```

Comment:
Perhaps add a test with a  (structured) quantity and a regular array - which should get dimensionless units.

Reply:
This opens up complications with structured regular arrays.
``arr = rfn.merge_arrays((q_pv["p"], q_pv.value), flatten=False)`` will fail because ``q_pv.value`` won't get a structured unit.

Reply:
``_as_quantities`` would have to be rewritten to construct a structured unit.

Reply:
A yes, a structured plain array would not work. What I meant to suggest was to add `q_pv['v'].value` or some other regular array.

We probably should change the behaviour of `Quantity` that when it gets a structured array, the default would be a `StructuredUnit` (all dimensionless) rather than plain `u.one`.  Probably as part of a larger effort to remove the possibility of having a structured value with an unstructured unit...

Reply:
For the git trail: #13676